src/EGraphs/egraph.jl

```diff
@@ -119,13 +115,15 @@ mutable struct EGraph{ExpressionType,Analysis}
   uf::UnionFind
   "map from eclass id to eclasses"
   classes::Dict{IdKey,EClass{Analysis}}
+  "vector of the original e-nodes"
+  nodes::Vector{VecExpr}
```

Comment:
I wonder if later on we can figure out something more efficient than vectors

Reply:
For this the vector is quite natural. 
New enodes are only pushed at the end, and we can simply index enodes by their non-canonical index.